### PR TITLE
Retries each record in loop with its own try catch [#120122659]

### PR DESCRIPTION
**Pull request by genericname92**

### CODE REVIEW

File: src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java [hunk 1]
```diff
@@ -38,23 +38,38 @@ public void kinesisHandler(KinesisEvent event, Context context){
...
-                firehoseClient.putRecord(putRecordRequest);
-                log.info("Successful Put");
+                putRecord(rec, 0);
             }
         } catch (Throwable e) {
             log.error("Kinesis handler threw exception", e);
         }
     }
 
+    private boolean putRecord(KinesisEvent.KinesisEventRecord rec, int tries) {
```
Comment (ajcantu) [inline src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java]: what's the point of the return value?

Comment (genericname92) [inline src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java]: hmm I guess nothing right now. I wasn't sure if we should log anything different for a failed vs success record. Something like if (!putRecord(rec, 0) { log.info("Failed Put") }

File: src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java [hunk 2]
```diff
@@ -38,23 +38,38 @@ public void kinesisHandler(KinesisEvent event, Context context){
...
+        PutRecordRequest putRecordRequest = new PutRecordRequest()
+        .withDeliveryStreamName(deliveryStreamName)
+        .withRecord(deliveryStreamRecord);
+
+        log.info("Putting message");
+        firehoseClient.putRecord(putRecordRequest);
+        log.info("Successful Put");
+        return true;
+      } catch (Throwable e) {
+        log.error("Kinesis putRecord failed on "+ new String(rec.getKinesis().getData().array()));
```
Comment (ajcantu) [inline src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java]: should we add tries & MAX_COUNT to the message here?

File: src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java (hunk 1 above)
Comment (ajcantu) [inline src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java]: it should be covered by the logging on line 63, no?

Comment (genericname92) [inline src/main/java/com/amazonaws/proserv/lambda/KinesisToFirehose.java]: yeah that's true, I'll turn it to void then

**Comment by ajcantu** [comment]
👍

**Pull request merged**